src/cmd/cmd_controller.js

```diff
@@ -226,12 +226,14 @@ class EmbarkController {
         callback();
       },
       function buildOrBuildAndDeploy(callback) {
-        if (options.onlyCompile) {
-          engine.events.request('contracts:build', {}, err => callback(err));
-        } else {
-          // deploy:contracts will trigger a build as well
-          engine.events.request('deploy:contracts', err => callback(err));
-        }
+        if (options.onlyCompile)
+          return engine.events.request('contracts:build', {}, err => {
+            if(err !== undefined) return callback(err);
```

Comment:
Maybe interesting for future changes: In JavaScript `undefined` and `null` are falsy values, so the check can be done with `if (err) { ... }`